postgres/assets/configuration/spec.yaml

```diff
@@ -570,6 +570,13 @@ files:
           value:
             type: string
             example: us-east-1
+        - name: iam_auth
+          description: |
+            Set to `true` to enable IAM authentication. 
+            `region` must be set to configure IAM authentication.
```

Comment:
we don't region for anything else, should it just be part of this block? thats what we do for MI https://github.com/DataDog/integrations-core/blob/master/postgres/datadog_checks/postgres/data/conf.yaml.example#L525-L539

Reply:
You mean make `managed_identity` to work for both Azure and AWS?

Reply:
I'm a bit hesitant to add aws iam auth to `managed_identity` because the term `Managed Identity` is very Azure specific. I'm now thinking have `managed_authentication` block for every cloud provider so that
1. it's clear that `managed_authentication` is associated with specific cloud provider
2. easy to enable `managed_authentication` for other cloud provider like gcp or easy to add new config options for existing cloud provider in the future

```yaml
aws:
  region: us-west-1
  managed_authentication:
    enabled: true # Enable IAM Authentication for AWS if true
    profile: <aws_credential_profile>  # example of new config option for the future

azure:
  # General Azure configuration...
  managed_authentication:
    enabled: true # Enable Managed Identity for Azure if true
    client_id: <azure_client_id>
    identity_scope: <identity_scope>

gcp:
  project_id: <gcp_project_id>
  instance_id: <gcp_instance_id>
```